tests/jit/test_jit_cache_providers.py

```
@@ -0,0 +1,558 @@
+        monkeypatch.setattr(module.platform, "machine", lambda: "x86_64")
+        monkeypatch.setattr(module.platform, "libc_ver", lambda: ("glibc", "2.28"))
+        monkeypatch.setenv(module.PROVIDER_PLATFORM_TAG_ENV, "manylinux_2_28_x86_64")
+        assert module._provider_platform_tag("linux_x86_64") == "manylinux_2_28_x86_64"
+
+        monkeypatch.setenv(module.PROVIDER_PLATFORM_TAG_ENV, "manylinux_2_28_aarch64")
+        with pytest.raises(RuntimeError, match="Unsupported provider platform tag"):
+            module._provider_platform_tag("linux_x86_64")
+
+        monkeypatch.setenv(module.PROVIDER_PLATFORM_TAG_ENV, "manylinux_2_28_x86_64")
+        monkeypatch.setattr(module.platform, "libc_ver", lambda: ("glibc", "2.34"))
+        with pytest.raises(RuntimeError, match="glibc 2.34 is too new"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)